src/zookeeperLock.ts

```diff
@@ -452,8 +600,28 @@ export class ZookeeperLock extends EventEmitter {
     };
 
 
-    private checkBail = () => {
-        return this.timedOut || this.destroyed || (this.client && (<any>this.client.getState()).name !== 'SYNC_CONNECTED');
+    /**
+     * are we on the happy path to continue locking?
+     * @returns {boolean|zk.Client}
+     */
+    private continueLocking = () => {
+        return this.state === ZookeeperLock.States.LOCKING && (this.client && (<any>this.client.getState()).name === 'SYNC_CONNECTED');
```

Comment:
CONNECTED is actually wrong, https://github.com/alexguan/node-zookeeper-client/blob/master/lib/State.js#L62 i'll remove it from the other place